client/bio-commandline/src/main/kotlin/uk/ac/ebi/biostd/client/cli/BioStudiesCommandLine.kt

```diff
@@ -1,83 +1,17 @@
-    private val server by option("-s", "--server", help = "BioStudies host url").required()
-    private val user by option("-u", "--user", help = "User that will perform the submission").required()
-    private val password by option("-p", "--password", help = "The user password").required()
-    private val onBehalf by option("-b", "--onBehalf", help = "Perform the operation on behalf of this user")
-    private val input by option(
-        "-i", "--input", help = "Path to the file containing the submission page tab").file(exists = true)
-    private val attached by option(
-        "-a", "--attached", help = "Comma separated list of paths to the files referenced in the submission")
-
-    @Suppress("TooGenericExceptionCaught")
+class BioStudiesCommandLine : CliktCommand() {
     override fun run() {
-        runCatching {
-            val files: MutableList<File> = mutableListOf()
-
-            attached?.let {
-                it.split(FILES_SEPARATOR).forEach { path -> addFiles(files, path) }
-            }
-
-            val client = getClient(server, user, password, onBehalf)
-            val submission = client.submitSingle(input!!, files).body
-
-            echo("SUCCESS: Submission with AccNo ${submission.accNo} was submitted")
-        }.onFailure {
-            val message = when (it) {
-                is WebClientException -> formatErrorMessage(it.message!!)
-                is SecurityWebClientException -> it.message!!
-                else -> it.message!!
-            }
-
-            throw PrintMessage(message)
-        }
-    }
-
-    internal fun getClient(host: String, user: String, password: String, onBehalf: String?) =
-        if (onBehalf.isNullOrBlank()) SecurityWebClient.create(host).getAuthenticatedClient(user, password)
-        else SecurityWebClient.create(host).getAuthenticatedClient(user, password, onBehalf)
-
-    private fun addFiles(files: MutableList<File>, path: String) {
-        val file = File(path)
-
-        if (file.isDirectory) file.walk().filter { it.isFile }.forEach { files.add(it) }
-        else files.add(file)
-    }
-
-    private fun formatErrorMessage(message: String): String {
-        val xmlOutput = StreamResult(StringWriter())
-        val xmlInput = StreamSource(StringReader(message))
-        val transformer =
-            TransformerFactory
-                .newInstance()
-                .apply { setAttribute("indent-number", 2) }
-                .newTransformer()
-
-        transformer.setOutputProperty(OutputKeys.INDENT, "yes")
-        transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes")
-        transformer.transform(xmlInput, xmlOutput)
-
-        return xmlOutput.writer.toString()
+        echo("That's BioStudies command line")
     }
 }
 
-fun main(args: Array<String>) = BioStudiesCommandLine().main(args)
+fun main(args: Array<String>) = BioStudiesCommandLine()
+    .subcommands(SubmitCommand(SubmissionService()), DeleteCommand(SubmissionService()))
```

Comment:
Just out of curiosity, couldn't we define a single instance of `SubmissionService` and pass it down to both sub commands? Would this give us any advantage?